internal/emigration/emigrate-nobles.lua

```diff
@@ -0,0 +1,352 @@
+--@module = true
+
+--Deport resident nobles of other lands and summon your rightful lord if he/she is elsewhere
+
+--[[
+TODO:
+  * Feature: have rightful ruler immigrate to fort if off-site
+  * QoL: make sure items are unassigned
+]]--
+
+local argparse = require("argparse")
+
+local options = {
+    all = false,
+    unitId = -1,
+    list = false
+}
+
+local fort = nil      ---@type df.world_site
+local playerCiv = nil ---@type df.historical_entity
+local capital = nil   ---@type df.world_site
```

Comment:
code that uses globals is harder to follow. Could you refactor this so they are passed as parameters? maybe as a `ctx` table so they don't need to be passed individually.